test/alternator/conftest.py

```diff
@@ -1,7 +1,7 @@
 # Copyright 2019-present ScyllaDB
 #
 # SPDX-License-Identifier: AGPL-3.0-or-later
-
+import allure
```

Comment:
Oh no :-( Please, find a way not to need adding all this code to test/*/conftest.py :-(
Can we call pytest with an extra parameter to load this extra code?

What happens when you run pytest manually or run test/alternator/run - does this allure stuff change anything in their behavior?

Reply:
Ok, I read some more and it appears that after you install allure-pytest (which it seems you already did) and use "--alluredir", allure is enabled. You don't need extra code in all the conftest.py.
It seems you added all that extra code just to set the "mode" and "run_id" thing to allure - but... you already did this in the past for pytest, so why do you need to do it specifically for allure?

Whatever you do here, I don't want test/cql-pytest/run or manually running pytest to depend on allure, which is only for CI and developers don't care about. At the minimum this code should be a no-op if allure isn't installed or "--alluredir" is not passed, and not just fail. But even better would be not to have this extra code at all. It's sad that we need allure to run a per-test autouse fixture to set exactly the same "runid" and "mode" parameters on all of them - isn't there some global option to do that on all the tests?

Reply:
allure it is a test reporting system and not only for CI, it can be used for local runs as well.  

now test.py has many problems with test reporting due to many versions of running the same test cases (modes, repeats) and custom XML reports buildings that have bugs, 
Allure can solve all current problems, it is a customizable, easy to integrate and what is important, tested solution.

Reply:
I have no use for allure for local runs, and definitely anybody who does not run the tests via test.py - e.g., test/cql-pytest/run-cassandra or `cd cql-pytest; pytest` will not use it, so I'm happy about the new version that makes it optional (and less intrusive in the code, as well).